mirage/scenarios/collections.ts

```diff
@@ -133,71 +151,338 @@ export function collectionModerationScenario(server: Server, currentUser: ModelI
+    projectBuilderArgument.server.create('contributor', {
         node,
-        users: project.currentUser,
+        users: projectBuilderArgument.currentUser,
         index: 0,
     });
-    project.server.create('collection-submission', {
-        creator: project.currentUser,
+
+
+    return projectBuilderArgument.server.create('collection-submission', {
+        creator: projectBuilderArgument.currentUser,
         guid: node,
         id: node.id,
         reviewsState,
-        collection: project.collection,
+        collection: projectBuilderArgument.collection,
+        submissionAction: projectBuilderArgument.collectionSubmissionActionArgument,
     });
 }
 
 /**
- * pendingProject
+ * chaosProject
  *
- * @description Abstracted function to easily build a pending project
- * @param project The project model
+ * @description Abstracted function to easily build a chaos project
+ * @param projectBuilderArgument The project builder argument
  */
-function pendingProject(project: ProjectModel) {
-    projectBuilder(project, CollectionSubmissionReviewStates.Pending);
-}
+function chaosProject(projectBuilderArgument: ProjectBuilderArgument): void {
+    const collectionSubmission = projectBuilder(projectBuilderArgument, CollectionSubmissionReviewStates.Accepted);
 
-/**
- * removedProject
- *
- * @description Abstracted function to easily build a removed project
- * @param project The project model
- */
-function removedProject(project: ProjectModel) {
-    projectBuilder(project, CollectionSubmissionReviewStates.Removed);
-}
+    projectBuilderArgument.server.create('collection-submission-action', {
+        fromState: CollectionSubmissionReviewStates.InProgress,
+        toState: CollectionSubmissionReviewStates.Pending,
+        actionTrigger: CollectionSubmissionActionTrigger.Submit,
+        target: collectionSubmission,
+        creator: projectBuilderArgument.currentUser,
+        dateCreated: getMinusDate(365),
+    });
 
-/**
- * rejectedProject
- *
- * @description Abstracted function to easily build a rejected project
- * @param project The project model
- */
-function rejectedProject(project: ProjectModel) {
-    projectBuilder(project, CollectionSubmissionReviewStates.Rejected);
+    projectBuilderArgument.server.create('collection-submission-action', {
+        fromState: CollectionSubmissionReviewStates.Pending,
+        toState: CollectionSubmissionReviewStates.Rejected,
+        actionTrigger: CollectionSubmissionActionTrigger.Reject,
+        target: collectionSubmission,
+        comment: 'I do not like your project.',
+        creator: projectBuilderArgument.currentUser,
+        dateCreated: getMinusDate(355),
+    });
+
+    projectBuilderArgument.server.create('collection-submission-action', {
+        fromState: CollectionSubmissionReviewStates.InProgress,
+        toState: CollectionSubmissionReviewStates.Pending,
+        actionTrigger: CollectionSubmissionActionTrigger.Submit,
```

Comment:
Since this was rejected earlier, these submission-action attributes should be 
```suggestion
        fromState: CollectionSubmissionReviewStates.Rejected,
        toState: CollectionSubmissionReviewStates.Pending,
        actionTrigger: CollectionSubmissionActionTrigger.Resubmit,
```

Same for line 364-366, 414-416, 434-436, and 454-456